zebrad/src/components/mempool/gossip.rs

```diff
@@ -0,0 +1,50 @@
+//! A task that gossips [`transaction::UnminedTxId`] that enter the mempool to peers.
+
+use tower::{timeout::Timeout, Service, ServiceExt};
+
+use zebra_network as zn;
+
+use tokio::sync::watch;
+use zebra_chain::transaction::UnminedTxId;
+
+use std::collections::HashSet;
+
+use crate::BoxError;
+
+use crate::components::sync::TIPS_RESPONSE_TIMEOUT;
+
+/// Run continuously, gossiping new [`transaction::UnminedTxId`] to peers.
+///
+/// Broadcast any [`transaction::UnminedTxId`] that gets stored in the mempool to all ready peers.
+pub async fn gossip_mempool_transaction_id<ZN>(
+    mut receiver: watch::Receiver<Option<UnminedTxId>>,
+    broadcast_network: ZN,
+) -> Result<(), BoxError>
+where
+    ZN: Service<zn::Request, Response = zn::Response, Error = BoxError> + Send + Clone + 'static,
+    ZN::Future: Send,
+{
+    info!("initializing transaction gossip task");
+
+    // use the same timeout as tips requests,
+    // so broadcasts don't delay the syncer too long
+    let mut broadcast_network = Timeout::new(broadcast_network, TIPS_RESPONSE_TIMEOUT);
+
+    loop {
+        // once we get new data in the channel, broadcast to peers
+        receiver.changed().await?;
+
+        let tx = receiver.borrow().unwrap();
```

Comment:
Testing: Is there a test that makes sure that this `unwrap` never panics?
Error Messages: Can we use `expect` here?

Reply:
I think sending a `HashSet` through the channel might remove the `unwrap`. Because we can just skip the broadcast if there are no transaction IDs.

See my other comment:
https://github.com/ZcashFoundation/zebra/pull/2823#discussion_r721817189